src/main/java/org/assertj/core/api/recursive/comparison/RecursiveComparisonDifferenceCalculator.java

```diff
@@ -689,7 +690,8 @@ private static boolean expectedTypeIsNotSubtypeOfActualType(DualValue dualField)
     return expectedTypeIsNotSubtypeOfActualType(dualField.actual, dualField.expected);
   }
 
-  private static boolean expectedTypeIsNotSubtypeOfActualType(Object actual, Object expected) {
+  private static boolean expectedTypeIsNotSubtypeOfActualType(@Nullable Object actual, @Nullable Object expected) {
+    //noinspection ConstantConditions
     return !actual.getClass().isAssignableFrom(expected.getClass());
```

Comment:
This is a potential NPE I found. Code that calls this method has nullable values. I added the `//noinspection`, but a better solution would be to handle the nulls